json_encoder.go

```diff
@@ -144,8 +159,12 @@ func (enc *jsonEncoder) AddObject(key string, obj interface{}) error {
 
 // Clone copies the current encoder, including any data already encoded.
 func (enc *jsonEncoder) Clone() encoder {
-	clone := newJSONEncoder()
+	clone := jsonPool.Get().(*jsonEncoder)
```

Comment:
The constructor also sets the message/time/level formatters back to the default, which we don't want here. I could pass the current formatters in as options, but that'll allocate a slice for the variadic args.